include/swift/ABI/Executor.h

```diff
@@ -64,6 +64,12 @@ class ExecutorRef {
     return ExecutorRef(reinterpret_cast<uintptr_t>(actor) | IsDefaultActor);
   }
 
+  static ExecutorRef fromOpaque(void *ptr) {
+    return ExecutorRef((uintptr_t)ptr);
+  }
+
+  void *toOpaque() const { return (void *)Value; }
+
```

Comment:
Hmm.  Executors are about to turn into two-word values.  I'll keep looking, but we need to understand this.

Reply:
I see! This is really just to distinguish between main/global, so I can stop doing it this way and use a single bit instead.

Reply:
Okay, good.  We might want to make `DispatchQueue` conform to `Executor` in a similarly lightweight way, but we should be able to recreate the `ExecutorRef` value efficiently from a `dispatch_queue_t` reference, so a discriminator should be fine for that.

Reply:
I changed this to just use `1` and `2` for global/main executors. If/when we make `DispatchQueue` conform, then we can just store those pointers, knowing they'll never be those values, and recreate the `ExecutorRef` from them in `executorForEnqueuedJob`.